src/Components/Layout.tsx

```diff
@@ -1,596 +1,344 @@
+import { QuickSelect } from "./QuickSelect";
+import CalendarTodayIcon from "@material-ui/icons/CalendarToday";
+import KeyboardTabIcon from "@material-ui/icons/KeyboardTab";
+import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
+import RefreshIcon from "@material-ui/icons/Refresh";
+import "./Styling.css";
+import { MuiThemeProvider } from "@material-ui/core/styles";
+import { GlobalContext } from "./Constants";
+import {
+  DateRangeUI,
+  uiData,
+  dropdownData as dData,
+  DropdownData,
+  refreshData as rData,
+  RefreshData,
+  bodyConfig as bConfig,
+  BodyConfig,
+} from "./Types";
+import { style as themeStyle } from "./Style";
+
+interface Inputs {
+  onTimerEvent(): void;
+  onDateEvent(dates: Date[]): void;
+  dateFormatter?: Intl.DateTimeFormat;
+  theme?: any;
+  commonlyUsedText?: number[];
+  quickSelectTerms?: string[];
+  quickSelectIntervals?: number[];
+  relativeTerms?: string[];
+  relativeIntervals?: number[];
+  localeString?: string;
+  nowText?: string;
+  minimumYearValue?: number;
+  maximumYearValue?: number;
+  humanizer?: any;
+  setStoredRange?(dateRange: string[]): void;
+  storedRange: string[] | null;
+}
+
+export const Layout: React.FC<Inputs> = (props) => {
+  // GLOBAL
+  let defaults = useContext(GlobalContext);
+  defaults.classes = themeStyle();
+  defaults.commonlyUsedText =
+    props.commonlyUsedText || defaults.commonlyUsedText;
+
+  defaults.quickSelectIntervals =
+    props.quickSelectIntervals || defaults.quickSelectIntervals;
+
+  defaults.relativeIntervals =
+    props.relativeIntervals || defaults.quickSelectIntervals;
+
+  defaults.minimumYearValue =
+    props.minimumYearValue || defaults.minimumYearValue;
+
+  defaults.maximumYearValue =
+    props.maximumYearValue || defaults.maximumYearValue;
+
+  defaults.localeObj = {
+    localeString: props.localeString || defaults.localeObj.localeString,
+    dateFormatter: props.dateFormatter || defaults.localeObj.dateFormatter,
+    quickSelectTerms:
+      props.quickSelectTerms || defaults.localeObj.quickSelectTerms,
+    relativeTerms: props.relativeTerms || defaults.localeObj.relativeTerms,
+    nowText: props.nowText || defaults.localeObj.nowText,
+    humanizer: props.humanizer || defaults.localeObj.humanizer,
+  };
+
+  // STORAGE
+  const setStoredRange:
+    | ((dateRange: DateRange) => void)
+    | (() => null) = props.setStoredRange
+    ? (dateRange) => {
+        let dates: string[] = [
+          String(dateRange.finalDates[0].getTime()),
+          String(dateRange.finalDates[1].getTime()),
+        ];
+        for (let i = 0; i < dates.length; i++) {
+          if (dateRange.isAbsolute(i)) {
+            dates[i] = dates[i] + "A";
+          } else {
+            dates[i] = dateRange.relativeMS[i] + "R";
+          }
+        }
+        return props.setStoredRange!(dates);
+      }
+    : () => {
+        return null;
+      };
+
+  let storedRange = new DateRange(defaults.localeObj, uiData);
+  {
```

Comment:
Odd start-of-block.  Probably unnecessary.

Reply:
Done